### DISCUSSION TITLE
Keycloak account console PatternFly design system alignment

### DISCUSSION THREAD

**Discussion by tyandor** [keycloak/keycloak#9923] [Archive]
- **Status**: Final
  
## Motivation

This proposal is to update the Keycloak Account Console to the latest version of the PatternFly design system ([release 2021.15](https://www.patternfly.org/v4/developer-resources/release-notes#2021.15-release-notes-2021-11-17)) and to follow the latest PatternFly design guidelines. This also brings the account console into visual / design alignment with the latest updates to the admin console.

## Implementation details

The most obvious visual changes applied to all pages are:

- Page content areas use a white background rather than white cards on a grey background (except for landing page)
- Left aligning content
- Updating font to Red Hat Text (body copy) and Red Hat Display (headings)
- Using sentence case for all text
  
Global code changes include: 

- @patternfly/react libraries updated to latest versions
- Fixes for breaking changes from the library updates 
- Layout and styling updates to accommodate designs

### Landing page

![](https://lh3.googleusercontent.com/rcX5WOXih_E0HWrilJhY88Li4OdX6Qxv2pc9r0I8TGsPxYAn2hYBmq4MXCrI22sgxtORmbjj8w039Gl4HlgKOBXJk9wN0rwgpiEowJQokTJh4lqZmHuJ2Tak1Zhx5hFxB73eUTiy)

- Updated to [PatternFly’s card view layout](https://www.patternfly.org/v4/demos/card-view)

### Personal info

![](https://lh5.googleusercontent.com/3WLTJYyKXCpLk_W86SDWFg3en_81eF0o2-WXEkVyllOAmH7QoUOBwQ0izmE2jUSocwnY-_hgD70dnPjK7rH6R0zD2vRbw4lnhGE4SzOL7W2rKK5AzOYFX1X5MImChEWhK5wQg20l)

- Defaulting to stacked labels and inputs for forms rather than a horizontal layout
- Removed examples from subheading text to allow for more flexibility
  
### Signing in

![](https://lh3.googleusercontent.com/YGjFgdG5rubldHUNolIQ4lz3mqAN-xDn2zMpwG1ujHdnNWl_Qhuo_tPa5no3yAlTtBZYU58moZZISFVVCG1vTraR5uw5vOsaqgXPLrMQD0HDSKESGbhmUac8_kOrseijHXyvi4wx)

- Button label change from “Update” to “Change password”
- Improved spacing for easier parsing of the page sections
  
### Device activity

![](https://lh6.googleusercontent.com/o35a7dPvksvmimxkk1PIB5H1z-GnkGkBrFkX8M9J4sbkRSmdIEJGAk9vzs17GOClRYoa1WEIrFLPyvoUtLmHPGNNu2MAikcQEsquCfGjAh6objR_O1MU11MvrqQV0oKQ_HVA4Oq5)

- Changed icon and row heading to focus on the device rather than the browser
- Used a [PatternFly description list](https://www.patternfly.org/v4/components/description-list) to improve readability within each row
- Added “Refresh” label to refresh icon
  
### Linked accounts

![](https://lh3.googleusercontent.com/-gNQJwcwFjVsTZMVOXAE5Zjepe_jReOH5rRxrWAhScRzbHpW4aq0dtQxvPV1oGqSCdY9f1WKwRBiXWTroexLqjBfaKBYeIvmG09ay9fsWdFrewkYTDNUHVXQ4dBc6AkeK_xD1RUa)

- Updated to latest [PatternFly label](https://www.patternfly.org/v4/components/label) style

### Applications

![](https://lh5.googleusercontent.com/djIrrfC3ClxzKDGER7phkW591GRFjQ3WM2ll3kapCz5P_YagXUOlmStGCc0Olh-b9Yie_5je1JZrEV-eyzYdJOljfmtYwEaIz9pjyiRxm3N0tcLqxMCxKJSPn5bBAAuSUKnrpuTr)

- Improved alignment in data list and expansion
- Added subheading for consistency with other pages

### Resources

![](https://lh4.googleusercontent.com/TgJim20v44IZAIorhpybxmPKrVxDV81T4P5v9VuCHwlSL-LrfcNHr34XFUJSdLQSxkk_ZKKBUoZBQi5Ddg2NvX6IkuRNmPQJZKnsfFAPueQRba9au_p9JE5WLukfHHikaWpraVxR)

- Updated to latest PatternFly tabs and filter components

**Comment by stianst** [comment]
@ssilvert @keycloak/keycloak-ui-ux FIY

**Comment by ssilvert** [comment]
I've reviewed this already and I'm happy with the design.
